apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java

```diff
@@ -0,0 +1,173 @@
+			codeGrp.setCodeSetDescTxt(request.getCodeSetDescTxt());
+			codeGrp.setCodeSetNm(request.getValueSetNm());
+			codeGrp.setLdfPicklistIndCd(request.getLdfPicklistIndCd());
+			// Add ValueSet
+			Codeset valueSet = new Codeset(asAdd(request));
+			CodesetId id = new CodesetId();
+			id.setClassCd(ValueSetConstants.CREATE_CLASS_CD);
+			id.setCodeSetNm(codeSetName);
+			valueSet.setId(id);			
+			valueSet.setCodeSetGroup(codeGrp);
+			// save changes
+			Codeset resultCodeSet = valueSetRepository.save(valueSet);
+			codeSetGrpMetaRepository.save(codeGrp);
+			// Add any accompanying Value Concepts
+			
+			if(request.getValues() != null) {
+			for (ValueSetRequest.CreateCodedValue valueConcept : request.getValues()) {
+				codeValueRepository.save(createValueConcept(valueConcept));
+
+			}
+			}
+			CreateValueSetResponse.ValueSetCreateShort body = new CreateValueSetResponse.ValueSetCreateShort();
+			body.setId(resultCodeSet.getId());
+			body.setAddTime(resultCodeSet.getAddTime());
+			body.setAddUserId(resultCodeSet.getAddUserId());
+			body.setValueSetNm(resultCodeSet.getValueSetNm());
+			response.setBody(body);
+			response.setMessage(ValueSetConstants.SUCCESS_MESSAGE);
+			response.setStatus(HttpStatus.CREATED);
+
+		} catch (Exception e) {
+			response.setMessage(e.getMessage());
+			response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+
+		}
+		return response;
+
+	}
+
+	public boolean checkValueSetNameExists(String codeSetNm) {
+		return (codeSetNm != null && valueSetRepository.checkValueSetName(codeSetNm) > 0);
+
+	}
+
+	public boolean checkCodeSetGrpMetaDatEntry(String codeShrtDescTxt, String codeSetNm) {
+		return (codeShrtDescTxt != null && codeSetNm != null
+				&& codeSetGrpMetaRepository.checkCodeSetGrpMetaDatEntry(codeShrtDescTxt, codeSetNm) > 0);
+	}
+
+	public long getCodeSetGroupID() {
+		long maxGroupID = valueSetRepository.getCodeSetGroupCeilID();
+		if (maxGroupID > 0) {
+			maxGroupID = codeSetGrpMetaRepository.getCodeSetGroupMaxID() + 10;
+		} else {
+			maxGroupID = 9910;
+		}
+		return maxGroupID;
+	}
+
+	public ValueSetCommand.AddValueSet asAdd(final ValueSetRequest request) {
+		return new ValueSetCommand.AddValueSet(
+				request.getAssigningAuthorityCd(),
+				request.getAssigningAuthorityDescTxt(), 
+				request.getCodeSetDescTxt(), 
+				request.getEffectiveFromTime(),
+				request.getEffectiveToTime(), 
+				request.getIsModifiableInd(), 
+				request.getNbsUid(),
+				request.getSourceVersionTxt(), 
+				request.getSourceDomainNm(), 
+				request.getStatusCd(),
+				request.getStatusToTime(), 
+				request.getCodeSetGroup(), 
+				request.getAdminComments(),
+				request.getValueSetNm(), 
+				request.getLdfPicklistIndCd(), 
+				request.getValueSetCode(),
+				request.getValueSetTypeCd(), 
+				request.getValueSetOid(), 
+				request.getValueSetStatusCd(),
+				request.getValueSetStatusTime(),
+				request.getParentIsCd(), 
+				request.getAddTime(), 
+				request.getAddUserId());
```

Comment:
Probably don't want to allow the UI to specify which user to set as the "added by" user. Can use the `UserDetailsProvider` component to retrieve the current user form the context

Reply:
@mpeels updated